[Pages] How-to for custom branch aliases (#4132)
cloudflare/cloudflare-docs: content/pages/how-to/custom-branch-aliases.md

---
pcx-content-type: how-to
title: Adding custom branch aliases
---

# Adding custom branch aliases

> In this tutorial, you will learn how to add a custom branch alias (`staging.example.com`) that will point to a specific branch (`staging`) on your pages project.

Reviewer: Hello, this is a great addition to the docs, thank you! If I may, can I add some content notes to make this publish ready?

This intro paragraph has a 'what will the user do' sentence. Could you include a 'why the user will do this' and 'how the user will do this' sentence to flesh out the purpose of this guide?

{{<Aside type= "note">}}

> At the moment, this setup is only supported when using Cloudflare DNS.

Author: Added suggestion into latest commit.

> If you attempt to follow this tutorial using an external DNS provider, your custom alias will be sent to the production branch of the Pages project.

Author: Added suggestion into latest commit.

{{</Aside>}}

> Firstly, make sure that you've made a commit on the branch you'd like to set up an alias for - this will prompt a build in pages.

Reviewer: ```suggestion
First, make sure that you have made a commit on the branch you would like to set up an alias for. This will prompt a build in Pages.
```

Author: Added suggestion into latest commit.

Next, add a custom domain under your Pages project for `staging.example.com`.

> ![Pages Custom Domains Overview](../media/pages_custom_domain-1.png)

Reviewer: ```suggestion
![Follow the instructions below to access the custom domains overview in the Pages dashboard.](../media/pages_custom_domain-1.png)
```

Author: Added suggestion into latest commit.

To do this:

> 1.  Go to **Pages** [in the dashboard](https://dash.cloudflare.com/?to=/:account/pages/).
> 2.  Select your project.
> 3.  Click on **Custom domains** and then **Setup a custom domain**
> 4.  Input the domain you'd like to use as the custom branch alias, such as `staging.example.com`
> 5.  Click on **Continue** and then **Activate domain**

Reviewer: ```suggestion
1.  Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/login).
2. In Account Home, go to **Pages**.
3.  Select your project.
4.  Select **Custom domains** > **Setup a custom domain**.
5. Input the domain you would like to use as the custom branch alias, such as `staging.example.com`
6.  Select **Continue** > **Activate domain**
```

Author: Added suggestion into latest commit.

> ![Pages Custom Domains Activation](../media/pages_custom_domain-2.png)

Reviewer: ```suggestion
![After selecting your custom domain, you will be asked to activate it.](../media/pages_custom_domain-2.png)
```

Author: Added suggestion into latest commit.

> After doing this, you'll want to go into the [DNS settings](https://dash.cloudflare.com/?to=/:account/:zone/dns) for the

Reviewer: ```suggestion
After activating your custom domain, go to [DNS](https://dash.cloudflare.com/?to=/:account/:zone/dns) for the `example.com` zone and change the `staging` record's value to include our branch alias.
```

Reviewer: Is this sentence asking for the user to find the ('staging') Name and change the Content to the branch alias value (`staging.your-project.pages.dev`) ? It is best that the instructional wording match the wording of the dashboard to avoid confusion.

Author: Added suggestion into latest commit.

> `example.com` zone and change the `staging` record's value to include our branch alias.

Reviewer: ```suggestion
```

Author: Added suggestion into latest commit.

> In this instance, we'll want to change `your-project.pages.dev` to `staging.your-project.pages.dev`.

Reviewer: ```suggestion
In this instance, change `your-project.pages.dev` to `staging.your-project.pages.dev`.
```

Author: Added suggestion into latest commit.

---
pcx-content-type: how-to

> title: Adding custom branch aliases

Reviewer: ```suggestion
title: Add a custom domain to a branch
```

---

> # Adding custom branch aliases

Reviewer: same inquiry as before @WalshyDev

> # Adding custom branch aliases

Reviewer: ```suggestion
# Add a custom domain to a branch
```

> In this tutorial, you will learn how to add a custom branch alias (`staging.example.com`) that will point to a specific branch (`staging`) on your Pages project.

Reviewer 2: You're adding a custom domain rather than a branch alias. Branch aliases already exist, you're using that rather than making a new one.

Author: ```suggestion
In this tutorial, you will learn how to add a custom domain (`staging.example.com`) that will point to a specific branch (`staging`) on your Pages project.
```

This will allow you to have a custom domain that will always show the latest build for a specific branch on your Pages project.

{{<Aside type= "note">}}

Currently, this setup is only supported when using Cloudflare DNS.

If you attempt to follow this guide using an external DNS provider, your custom alias will be sent to the production branch of your Pages project.

{{</Aside>}}

> First, make sure that you have made a commit on the branch you would like to set up an alias for. This will prompt a build in Pages.

Reviewer 2: I'd change this to make sure you have at least 1 successful deployment.

builds won't exist for direct uploads

> Next, add a custom domain under your Pages project for `staging.example.com`.

Reviewer 2: for `staging.example.com` (Or what you'd want your custom subdomain to be).

---

Kate can probably give better wording...

Author: ```suggestion
Next, add a custom domain under your Pages project for your desired custom domain, i.e `staging.example.com`.
```
